### DISCUSSION TITLE
getServerSideProps & apollo data props hell.

### DISCUSSION THREAD

**Discussion by ITSWYoo** [vercel/next.js#14802] [Help]
### Page Home Component
```js
const Home = ({data}) => {
  <Gallery data={data} />
} 

export const getServerSideProps: GetServerSideProps<Props> = async (context: GetServerSidePropsContext) => {
  const { channelName: brandUrl, id } = context.params
  const apolloClient = initApollo()

  const productQueryResult: ApolloQueryResult<ProductQuery> = await apolloClient.query<
    ProductQuery,
    ProductQueryVariables
  >({
    query: ProductDocumentQuery,
    variables: {
      channelProductNo: Number(id),
      // @ts-expect-error
      brandUrl,
    },
  })

 // extract cache
  const initialState = apolloClient.cache.extract()

  return {
    props: {
      initialState,
      data: productQueryResult?.data,
      pageType: PageType.PRODUCT,
    },
  }
}
```
### Gallery component
```js
const Gallery =({data}) => {
  <Photo data={data}/>
}
```

`Home -> Gallery -> Photo`

In order to avoid deep and complex props delivery, I would like to access the data for ssr directly from the photo component rather than props. (like a redux but excluding the use of Context api, redux) 
If you have any ideas for this, please help. : )

**Comment by rafaelalmeidatk** [comment]
Hi! You can use the `with-apollo` example as reference since it does exactly what you are asking: https://github.com/vercel/next.js/blob/canary/examples/with-apollo
The posts list is fetched in `getStaticProps` [here](https://github.com/vercel/next.js/blob/canary/examples/with-apollo/pages/index.js#L23), then the cache is extracted into the `initialApolloState`. This prop will be utilized at `_app` [here](https://github.com/vercel/next.js/blob/canary/examples/with-apollo/pages/_app.js#L5) to populate the Apollo cache, so in the `PostList` component you do the same request [here](https://github.com/vercel/next.js/blob/canary/examples/with-apollo/components/PostList.js#L7-L37) and instead of doing another network request, it will use the cache that has the content fetched in `getStaticProps`

**Reply by ITSWYoo** [reply]
oh thx : )
I was worried about component re-rendering after SSR when using useQuery.
Since the data using the cache is the same, is there any performance problem even if it is re-rendered after SSR?

**Reply by ITSWYoo** [reply]
@rafaelalmeidatk 
Is graphql request only available within getserversideProps?

Can useQuery only fetch cache? in ssr
during ssr, If fetchpolicy: no-cache, can't graphql be called? useQuery in component

**Reply by msreekm** [reply]
you could pass the result of query in getserversideprops into the page in props ,with cache disabled, I have tested this..
you dont have to use cache 

 but what is not clear for me is if the query has pagination and you want to refetch the next page on click of button on page ,how do you get hold of refetch() method which is usually obtained from useQuery
